mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx

```diff
@@ -133,59 +134,6 @@ export const getDefaultSafetyFilter = (sheetDataArray: SheetData[], sheetIndex:
     return sheetDataArray[sheetIndex] === undefined || sheetDataArray[sheetIndex].numRows > GRAPH_SAFETY_FILTER_CUTOFF
 }
 
-/*
-    A helper function for getting the params for the graph for this sheet when
-    opening the graphing taskpane, or when switching to a sheet.
-
-    Notably, will filter oout any columns that are no longer in the dataset, 
-    which stops the user from having invalid columns selected in their graph
-    params.
-*/
-export const getGraphParams = (   
-    graphDataDict: GraphDataDict,
-    graphID: GraphID,
-    selectedSheetIndex: number,
-    sheetDataArray: SheetData[],
-    selectedColumnIds?: ColumnID[]
-): GraphParamsFrontend => {
-
-    const graphParamsCopy: GraphParamsFrontend = window.structuredClone(graphDataDict[graphID]?.graphParams); 
-
-    // If the graph already exists, get the data source sheet index from the graph params.
-    // Otherwise create a new graph of the selectedSheetIndex
-    const graphDataSourceSheetIndex = graphParamsCopy !== undefined ? graphParamsCopy.graphCreation.sheet_index : selectedSheetIndex
-
-    // If the graph already exists, retrieve the graph params that still make sense. In other words, 
-    // if a column was previously included in the graph and it no longer exists, remove it from the graph. 
-    if (graphParamsCopy !== undefined) {
-        // Filter out column headers that no longer exist
-        const validColumnIDs = sheetDataArray[graphDataSourceSheetIndex] !== undefined ? sheetDataArray[graphDataSourceSheetIndex].data.map(c => c.columnID) : [];
-        const xAxisColumnIDs = intersection(
-            validColumnIDs,
-            graphParamsCopy.graphCreation.x_axis_column_ids
-        )
-        const yAxisColumnIDs = intersection(
-            validColumnIDs,
-            graphParamsCopy.graphCreation.y_axis_column_ids
-        )
```

Comment:
This is a bug. We need to make sure to filter out invalid columns from the existing params. 

Check out how Pivot does it. It takes the existingParams in it's getDefaultPivotParams.

I think we can do the same for graph. If these existingParams are defined, then we can filter out invalid columns using this code.